src/libraries/System.Reflection.Metadata/src/System/Reflection/Internal/MemoryBlocks/StreamMemoryBlockProvider.cs

```diff
@@ -78,7 +76,7 @@ internal static unsafe NativeHeapMemoryBlock ReadMemoryBlockNoLock(Stream stream
 
                 int bytesRead = 0;
 
-                if (!isFileStream || (bytesRead = FileStreamReadLightUp.ReadFile(stream, block.Pointer, size)) != size)
+                if ((bytesRead = FileStreamReadLightUp.ReadFile(stream, block.Pointer, size)) != size)
```

Comment:
If we're removing "light up", does this class need to be renamed?

Reply:
Fixed. I renamed the method to `Read` and moved it to `StreamExtensions`.